source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+//                      spherical_representative_point[1] <= gm->maximum_longitude(),
+//                      ExcMessage("Longitude of representative point outside chunk domain."));
+
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          if (dim ==3)
+            {
+
+//              AssertThrow(spherical_representative_point[2] >= gm->minimum_latitude() &&
+//                          spherical_representative_point[2] <= gm->maximum_latitude(),
+//                          ExcMessage("Representative point outside chunk domain."));
+
+              spherical_representative_point[2] *= degrees_to_radians;
+            }
+
+          // TODO this is only correct for eccentricity zero
+          // or ask for the correct radius from user
+          spherical_representative_point[0] = gm->get_semi_major_axis_a();
+        }
+      else if (const GeometryModel::Sphere<dim> *gm = dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()))
+        {
+          spherical_representative_point[0] = gm->radius();
+          spherical_representative_point[1] *= degrees_to_radians;
+          spherical_representative_point[2] *= degrees_to_radians;
+        }
+      else if (const GeometryModel::Box<dim> *gm = dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model()))
+        {
+          const Point<dim> extents = gm->get_extents();
+
+          for (unsigned int d = 0; d < dim-1; ++d)
+            AssertThrow(representative_point[d] <= extents[d], ExcMessage("Coordinate "
+                                                                          + dealii::Utilities::int_to_string(d)
+                                                                          + " of representative point outside box domain."));
+
+          // set z of surface
+          representative_point[dim-1] = extents[dim-1];
+        }
+      else if (const GeometryModel::TwoMergedBoxes<dim> *gm = dynamic_cast<const GeometryModel::TwoMergedBoxes<dim>*> (&this->get_geometry_model()))
+        {
+          const Point<dim> extents = gm->get_extents();
+
+          for (unsigned int d = 0; d < dim-1; ++d)
+            AssertThrow(representative_point[d] <= extents[d], ExcMessage("Coordinate "
+                                                                          + dealii::Utilities::int_to_string(d)
+                                                                          +"Representative point outside box domain."));
+
+          // set z of surface
+          representative_point[dim-1] = extents[dim-1];
+        }
+      else
+        AssertThrow(false, ExcNotImplemented());
+
+      // set up the input for the density function of the material model
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in0(1, n_compositional_fields);
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out0(1, n_compositional_fields);
```

Comment:
This is just a suggestion, but I thought you could save some code by just reusing 1 MaterialModelInputs and 1 MaterialModelOutputs, and then you can refill it every time you need to evaluate, but you would only have to set `in.strain_rate.resize(0)` once, and you wouldn't have to generate new Inputs/Outpus objects inside of your loop over all cells.